_source/_posts/2019-12-30-java-jersey-jaxrs.adoc

```diff
@@ -0,0 +1,595 @@
+====
+
+To delete the task you're just removing it from the map, by passing the task ID.
+
+Let's test it! Run the application with the latest changes, then go to your terminal and type the following command:
+
+[source,text]
+----
+curl -X POST \
+  http://localhost:8080/tasks \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "description" : "do the dishes"
+}'
+----
+
+The command above will create a new task for you, with the description "do the dishes". Copy the ID of the task you just created and replace with `<task_id>` in the command above:
+
+[source,text]
+----
+curl -X DELETE http://localhost:8080/tasks/<task_id>
+----
+
+After you execute the command the task is going to be deleted. If you list your tasks again, you'll notice that the task is not there anymore.
+
+Now that you have a CRUD application up and running, the last step is to make sure only authenticated users can have access to it.
+
+== Secure Your Jersey REST Application
+
+You're going to use Okta to authenticate your users, so let's start by creating an account.
+
+If you don't have an Okta account, https://developer.okta.com/signup[go ahead and create one]. After creating it, go through the following steps:
+
+* Log in to your account
+* Click in **Applications**
+* Click on **Add Application**. You will be redirected to the following page:
+
+image::{% asset_path 'blog/java-jersey-jaxrs/okta-applications.png' %}[alt=Okta application page,width=600,align=center]
+
+* Select **Web** and click **Next**
+* Fill in the following options in the form:
+** Name: my-first-app
+** Base URIs: http://localhost:8080
+** Login redirect URL:
+*** `https://oidcdebugger.com/debug`
+** Grant Type allowed:
+*** Authorization Code
+*** Implicit(Hybrid)
+**** Allow Access Token with implicit grant type
+* Click **Done**
+
+Now that you have your Okta application you can use it to authenticate inside your app.
+
+== Secure Your Jersey Service
+
+Let's start by adding Okta's library inside your project.
+
+Go to the `pom.xml` and add the following dependency inside the `<depencencies>` tag:
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.okta.spring</groupId>
+    <artifactId>okta-spring-boot-starter</artifactId>
+    <version>1.3.0</version>
+</dependency>
+----
+
+This library will integrate with your Okta app you just created. It will also add Spring Security to your current application.
+
+Inside `src/main/java/com/okta/jerseyrest/configuration` create the following `SecurityConfiguration` class:
+
+[source,java]
+----
+package com.okta.jerseyrest.configuration;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+
+@Configuration
```

Comment:
This annotation is not necessary. It's included in [`EnableWebSecurity`](https://docs.spring.io/spring-security/site/docs/4.0.x/apidocs/org/springframework/security/config/annotation/web/configuration/EnableWebSecurity.html).